### PR TITLE
No instances error

### PR DESCRIPTION
**Pull request by joshski**
Also:

* Upgrade AWSSDK dependency
* Fix Nuget reference to NUnit
* Fix typo in AWSPushAndDeploly (Deploy)
* try/catch/explain when an exception happens
* Exit code 666 when an exception happens

### CODE REVIEW

File: TTC.Deployment.AmazonWebServices/NoInstancesException.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace TTC.Deployment.AmazonWebServices
+{
+    public class NoInstancesException : Exception
+    {
+        public override string Message
+        {
+            get
+            {
+                return "Found no instances to deploy to. Your application set directory should include subdirectories corresponding to instance tags specified in your VPC template";
```
Comment (PaulCampbell) [inline TTC.Deployment.AmazonWebServices/NoInstancesException.cs]: :+1:

File: AWSDeployCode/AWSPushAndDeploy.csproj [hunk 2]
```diff
@@ -50,15 +50,11 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="AWSSDK, Version=2.3.46.0, Culture=neutral, PublicKeyToken=9f476d3089b52be3, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\AWSSDK.2.3.46.0\lib\net45\AWSSDK.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="AWSSDK">
+      <HintPath>..\packages\AWSSDK.2.3.51.0\lib\net45\AWSSDK.dll</HintPath>
```
Comment (Jetski5822) [inline AWSDeployCode/AWSPushAndDeploy.csproj]: Due to a bug in older versions of nuget, we need <Private>True</Private> set.

File: AWSDeployCode/AWSPushAndDeploy.csproj [hunk 3]
```diff
@@ -50,15 +50,11 @@
...
+    <Reference Include="AWSSDK">
+      <HintPath>..\packages\AWSSDK.2.3.51.0\lib\net45\AWSSDK.dll</HintPath>
     </Reference>
     <Reference Include="CommandLine">
-      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\packages\CommandLineParser.1.9.71\lib\net45\CommandLine.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Configuration" />
```
Comment (Jetski5822) [inline AWSDeployCode/AWSPushAndDeploy.csproj]: Revert this file

File: AWSProvisionCloud/AWSProvisionCloud.csproj [hunk 4]
```diff
@@ -50,9 +50,8 @@
...
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="AWSSDK, Version=2.3.46.0, Culture=neutral, PublicKeyToken=9f476d3089b52be3, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\AWSSDK.2.3.46.0\lib\net45\AWSSDK.dll</HintPath>
+    <Reference Include="AWSSDK">
+      <HintPath>..\packages\AWSSDK.2.3.51.0\lib\net45\AWSSDK.dll</HintPath>
     </Reference>
     <Reference Include="CommandLine">
       <HintPath>..\packages\CommandLineParser.1.9.71\lib\net45\CommandLine.dll</HintPath>
```
Comment (Jetski5822) [inline AWSProvisionCloud/AWSProvisionCloud.csproj]: Revert

File: AWSS3Push/AWSS3Push.csproj [hunk 5]
```diff
@@ -35,9 +35,8 @@
...
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="AWSSDK, Version=2.3.46.0, Culture=neutral, PublicKeyToken=9f476d3089b52be3, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\AWSSDK.2.3.46.0\lib\net45\AWSSDK.dll</HintPath>
+    <Reference Include="AWSSDK">
+      <HintPath>..\packages\AWSSDK.2.3.51.0\lib\net45\AWSSDK.dll</HintPath>
     </Reference>
     <Reference Include="CommandLine">
       <HintPath>..\packages\CommandLineParser.1.9.71\lib\net45\CommandLine.dll</HintPath>
```
Comment (Jetski5822) [inline AWSS3Push/AWSS3Push.csproj]: Revert

File: TTC.Deployment.AmazonWebServices/TTC.Deployment.AmazonWebServices.csproj [hunk 6]
```diff
@@ -32,9 +32,8 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="AWSSDK, Version=2.3.46.0, Culture=neutral, PublicKeyToken=9f476d3089b52be3, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\AWSSDK.2.3.46.0\lib\net45\AWSSDK.dll</HintPath>
+    <Reference Include="AWSSDK">
+      <HintPath>..\packages\AWSSDK.2.3.51.0\lib\net45\AWSSDK.dll</HintPath>
```
Comment (Jetski5822) [inline TTC.Deployment.AmazonWebServices/TTC.Deployment.AmazonWebServices.csproj]: Revert this change

File: TTC.Deployment.Tests/TTC.Deployment.Tests.csproj [hunk 7]
```diff
@@ -32,9 +32,8 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="AWSSDK, Version=2.3.46.0, Culture=neutral, PublicKeyToken=9f476d3089b52be3, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\AWSSDK.2.3.46.0\lib\net45\AWSSDK.dll</HintPath>
+    <Reference Include="AWSSDK">
+      <HintPath>..\packages\AWSSDK.2.3.51.0\lib\net45\AWSSDK.dll</HintPath>
     </Reference>
```
Comment (Jetski5822) [inline TTC.Deployment.Tests/TTC.Deployment.Tests.csproj]: Revert this

**Comment by Jetski5822** [comment]
@joshski  Seems like you came across the Nuget bug in earlier versions of Visual Studio. :( You need <private>true</private> to be set.

**Comment by PaulCampbell** [comment]
apart from the proj files, lgtm

**Comment by PaulCampbell** [comment]
You'll need to package it and push to nuget too. @DaveOps83 has the login details

File: AWSDeployCode/AWSPushAndDeploy.csproj (hunk 3 above)
Comment (joshski) [inline AWSDeployCode/AWSPushAndDeploy.csproj]: I will not!

File: TTC.Deployment.AmazonWebServices/NoInstancesException.cs (hunk 1 above)
Comment (PaulCampbell) [inline TTC.Deployment.AmazonWebServices/NoInstancesException.cs]: Maybe add the tag name to this message? DeployRole I believe?

**Comment by Jetski5822** [comment]
GTG
